### PR TITLE
Reduce execution time by a further 60%

### PR DESCRIPTION
**Pull request by XertroV**
Mostly this is just using handles instead of instantiating new objects and avoiding calling the same functions multiple times.
Some other minor optimizations like reorder some maths operations and left padding

### CODE REVIEW

File: DID.as [hunk 1]
```diff
@@ -52,20 +56,21 @@ namespace DID {
...
+        // copying to leftPadded is cheap enough to be negligiable performance wise
         for (uint i = 0; i < 4; i++) {
-            leftPadded[i] = lanes[i].content;
-            while (leftPadded[i].Length < maxLen) {
-                leftPadded[i] = " "+leftPadded[i]; // str_pad when 🥺
-            }
+            leftPadded[i] = LOOOOOOOONG.SubStr(0, maxLen - lanes[i].content.Length) + lanes[i].content;
+            // while (leftPadded[i].Length < maxLen) {
+            //     leftPadded[i] = " "+leftPadded[i]; // str_pad when 🥺
+            // }
```
Comment (sylae) [inline DID.as]: lose this commented code pls if we're not using it

File: DID.as [hunk 2]
```diff
@@ -52,20 +56,21 @@ namespace DID {
...
         }
 
         int leftOffset = (diegeticHorizontalDistance + maxLen*100) * -1;
         int rightOffset = diegeticHorizontalDistance;
         for (uint i = 0; i < 4; i++) {
-            LaneConfig@ left = lanes[i];
-            LaneConfig@ right = lanes[i+4];
+            // LaneConfig@ left = lanes[i];
+            // LaneConfig@ right = lanes[i+4];
```
Comment (sylae) [inline DID.as]: yeet this pls

File: DID.as [hunk 3]
```diff
@@ -81,14 +86,14 @@ namespace DID {
...
+        }
     }
 
     DID::LaneConfig@[] lanes;
-    dictionary font;
+
 
     void Main() {
-        init();
+        OnSettingsChanged();
```
Comment (sylae) [inline DID.as]: for my sanity i prefer to keep all this stuff in a distinct init() called both at plugin startup and when things change that require reloading that kinda thing

File: DID.as [hunk 4]
```diff
@@ -104,6 +109,7 @@ namespace DID {
 
     void OnSettingsChanged() {
         init();
+        SetLaneProvidersFromSettings();
```
Comment (sylae) [inline DID.as]: put in init() pls?

File: DID.as [hunk 5]
```diff
@@ -234,6 +246,25 @@ namespace DID {
...
 
+
+    dictionary font;
+    /**
+     * Only chars supported: [0-9+-.:/]
+     * ascii: 0x30-39 for 0-9
+     * 0x2b for +, then ,-./
+     * 0x3a for :
+     */
+    // int fontCharOffset = 0x2b;
```
Comment (sylae) [inline DID.as]: unused vars... generally my philosophy is if i need old code i'll find it in the log or blame.

File: Glyph.as [hunk 6]
```diff
@@ -1,26 +1,37 @@
...
-                drawGlyph(cast<vec3[]>(dv), offset, z);
-            } else {
-                warn("no glyph for char "+l);
-            }
+            // if (font.Exists(l)) {
+            //     dictionaryValue dv = font[l];
+            //     drawGlyph(cast<vec3[]>(dv), offset, z);
+            // } else {
+            //     warn("no glyph for char "+l);
+            // }
```
Comment (sylae) [inline Glyph.as]: kill

File: SpaceStuff.as [hunk 7]
```diff
@@ -1,7 +1,20 @@
...
+        @vis = _vis;
+    }
+
+    // EXPENSIVE!!!!!!!!
+    // CSceneVehicleVis@ vis {
+    //     get {
+    //         return VehicleState::GetVis(GetApp().GameScene, VehicleState::GetViewingPlayer());
+    //     }
+    // }
+
```
Comment (sylae) [inline SpaceStuff.as]: you know the drill :p

File: SpaceStuff.as [hunk 8]
```diff
@@ -29,32 +43,35 @@ namespace DID {
...
+        return true;
 
-        if (c1_3.z < 0 && c2_3.z < 0 && d_3.z < 0) {
-            nvg::BezierTo(c1_3.xy, c2_3.xy, d_3.xy);
-            return true;
-        }
-        return false;
+        // if (c1_3.z < 0 && c2_3.z < 0 && d_3.z < 0) {
+        // }
+        // return false;
```
Comment (sylae) [inline SpaceStuff.as]: kill it

File: SpaceStuff.as [hunk 9]
```diff
@@ -29,32 +43,35 @@ namespace DID {
...
     }
 
     // stolen from hats mod
-    vec3 projectHatSpace(vec3 point) {
-        auto visState = VehicleState::GetVis(GetApp().GameScene, VehicleState::GetViewingPlayer()).AsyncState;
-        vec3 localPos = (visState.Left * point.x) + (visState.Up * point.y) + (visState.Dir * point.z);
-        return visState.Position + localPos + visState.Dir + visState.Up;
+    vec3 projectHatSpace(const vec3 &in point) {
+        vec3 localPos = (vis.AsyncState.Left * point.x) + (vis.AsyncState.Up * (point.y + 1.)) + (vis.AsyncState.Dir * (point.z + 1.));
+        // auto l = vis.AsyncState.Left, u = vis.AsyncState.Up, d = vis.AsyncState.Dir;
```
Comment (sylae) [inline SpaceStuff.as]: :eyes:

**Review by sylae — CHANGES_REQUESTED** [review]
thanks for the PR! this all looks really nice, just have an excessive amount of me malding over code style stuff :p

File: DID.as (hunk 4 above)
Comment (XertroV) [reply DID.as]: I don't think it's needed in init. Will add it b/c the overhead is negligible

File: DID.as (hunk 3 above)
Comment (XertroV) [reply DID.as]: Moved things around, hopefully makes more sense now.

**Review by sylae — APPROVED** [review]

**Pull request merged**